ppl/lake/finder.go

```diff
@@ -102,25 +102,31 @@ func Find(ctx context.Context, zctx *resolver.Context, lk *Lake, query index.Que
 	}
 	return Walk(ctx, lk, func(chunk chunk.Chunk) error {
 		dir := chunk.ZarDir()
-		rec, err := index.Find(ctx, zctx, dir, matched.DefID, matched.Values...)
-		if rec == nil || (err != nil && zqe.IsNotFound(err) && opt.skipMissing) {
+		reader, err := index.Find(ctx, zctx, dir, matched.DefID, matched.Values...)
+		if err != nil && zqe.IsNotFound(err) && opt.skipMissing {
 			// No index for this rule.  Skip it if the skip boolean
 			// says it's ok.  Otherwise, we return ErrNotExist since
 			// the client was looking for something that wasn't indexed,
 			// and they would probably want to know.
 			return nil
 		}
 
-		if rec, err = opt.addPath(lk, chunk, rec); err != nil {
-			return err
-		}
+		defer reader.Close()
```

Comment:
The error handling needs adjustment here, unrelated to these changes.  There's a missing `if err != nil return` that should happen before this line, I think.

Reply:
Right you are.